stage23/protos/bootboot.c

```diff
@@ -101,34 +182,22 @@ void bootboot_load(char *config) {
         }
     }
 
-    printv("bootboot: mapping struct to %X", struct_vaddr);
-    printv("bootboot: mapping environemnt to %X", env_vaddr);
-    printv("bootboot: mapping framebuffer to %X", fb_vaddr);
-    printv("bootboot: the init stack is %X bytes", init_stack_size);
-
-    uint64_t entry, top, slide, rangecount, physbase, virtbase = 0;
-    struct elf_range* ranges;
-
-    /// Memory mappings ///
-    pagemap_t pmap = new_pagemap(4);
-
-    /// Load kernel ///
-    {
-        if (elf64_load(
-            kernel, &entry, &top, &slide, MEMMAP_KERNEL_AND_MODULES,
-            false, false, &ranges, &rangecount, true, &physbase, &virtbase)) {
-            panic("bootboot: elf64 load failed");
-        }
-        for (uint64_t mapvirt = virtbase, mapphys = physbase; mapphys < top;mapvirt += 0x1000, mapphys += 0x1000) {
-            map_page(pmap, mapvirt, mapphys, VMM_FLAG_PRESENT | VMM_FLAG_WRITE, false);
-        }
+    if (init_stack_size == ~0UL) {
+        print("bootboot: warning: no init stack size entered, assuming 1024\n");
```

Comment:
Suggestion: Maybe instead of making the stack too small, make it a reasonable about (like `4096 * 16`) and just warn that:
```
bootboot: WARNING: No init stack size specified, defaulting to `0x10000` (consult CONFIG.md). 
````

Also a note that warning will just be visible for a short second of time, maybe add a short delay so the warning is actually visible?

Reply:
The spec says 1024, so i put 1024. But i will add the delay

Reply:
Alrighty